src/Apps/W1/Subscription Billing/Test/Billing/AutomatedBillingTest.Codeunit.al

```diff
@@ -0,0 +1,203 @@
+        InitUserSetupWithoutAuthorizationForAutomatedBilling(UserSetup);
+
+        // [WHEN] User tries to create a billing template with automation
+        ContractTestLibrary.CreateDefaultRecurringBillingTemplateForServicePartner(BillingTemplate, BillingTemplate.Partner::Customer);
+        Commit(); // The Billing Template must be committed before validating the Automation field.
+
+        // [THEN] Setting automation should fail
+        asserterror BillingTemplate.Validate(Automation, BillingTemplate.Automation::"Create Billing Proposal and Documents");
+        Assert.ExpectedError(AutoContractBillingNotAllowedErr);
+
+        // [GIVEN] A user with Auto Contract Billing permission
+        InitUserSetupWithAuthorizationForAutomatedBilling(UserSetup);
+
+        // [WHEN] User tries to set automation
+        BillingTemplate.Validate(Automation, BillingTemplate.Automation::"Create Billing Proposal and Documents");
+        BillingTemplate.Modify(true);
+
+        // [THEN] Automation should be set successfully
+        BillingTemplate.TestField(Automation, BillingTemplate.Automation::"Create Billing Proposal and Documents");
+    end;
+
+    [Test]
+    procedure CreateJobQueueEntryForBillingTemplate()
+    var
+        JobQueueEntry: Record "Job Queue Entry";
+        BillingTemplate: Record "Billing Template";
+    begin
+        // [SCENARIO] Verify that a Job Queue Entry is created correctly for a Billing Template with automation settings
+        Initialize();
+
+        // [WHEN] Create a Billing Template with automation settings
+        CreateBillingTemplateWithAutomation(BillingTemplate);
+
+        // [THEN] Verify Job Queue Entry is created with correct parameters
+        JobQueueEntry.Get(BillingTemplate."Batch Recurrent Job Id");
+        Assert.AreEqual(JobQueueEntry."Object Type to Run"::Codeunit, JobQueueEntry."Object Type to Run", 'Object Type should be Codeunit.');
+        Assert.AreEqual(Codeunit::"Auto Contract Billing", JobQueueEntry."Object ID to Run", 'Object ID should be Auto Contract Billing codeunit.');
+        Assert.IsTrue(JobQueueEntry."Recurring Job", 'Job should be recurring');
+        Assert.AreEqual(BillingTemplate."Minutes between runs", JobQueueEntry."No. of Minutes between Runs", 'Minutes between runs should match.');
+    end;
+
+    [Test]
+    [HandlerFunctions('MessageHandler,ExchangeRateSelectionModalPageHandler')]
+    procedure RunAutomatedProcessingForNewBillingTemplate()
+    var
+        CustomerSubscriptionContract: Record "Customer Subscription Contract";
+        SubscriptionHeader: Record "Subscription Header";
+        BillingTemplate: Record "Billing Template";
+        BillingLine: Record "Billing Line";
+        JobQueueEntry: Record "Job Queue Entry";
+        SalesHeader: Record "Sales Header";
+    begin
+        // [SCENARIO] Verify that a Job Queue Entry is created correctly for a Billing Template with automation settings
```

Comment:
This scenario description looks wrong.

Reply:
This scenario description seems to be copied from another test. The description didnt' fit to the test.